kategory/src/test/kotlin/kategory/free/FreeApplicativeTest.kt

```diff
@@ -0,0 +1,40 @@
+package kategory
+
+import io.kotlintest.KTestJUnitRunner
+import io.kotlintest.matchers.shouldBe
+import org.junit.runner.RunWith
+
+sealed class OpsAp<out A> : HK<OpsAp.F, A> {
+
+    class F private constructor()
+
+    data class Value(val a: Int) : OpsAp<Int>()
+    data class Add(val a: Int, val y: Int) : OpsAp<Int>()
+    data class Subtract(val a: Int, val y: Int) : OpsAp<Int>()
+
+    companion object : FreeApplicativeInstances<F> {
+        fun value(n: Int): FreeApplicative<F, Int> = FreeApplicative.liftF(Value(n))
+        fun add(n: Int, y: Int): FreeApplicative<F, Int> = FreeApplicative.liftF(Add(n, y))
+        fun subtract(n: Int, y: Int): FreeApplicative<F, Int> = FreeApplicative.liftF(Subtract(n, y))
+    }
+}
+
+fun <A> HK<OpsAp.F, A>.ev(): OpsAp<A> = this as OpsAp<A>
+
+@RunWith(KTestJUnitRunner::class)
+class FreeApplicativeTest : UnitSpec() {
+
+    private val program = OpsAp.tupled(OpsAp.value(1), OpsAp.add(3, 4), OpsAp.subtract(3, 4)).ev()
+
+    init {
+
+        testLaws(ApplicativeLaws.laws(OpsAp, FreeApplicativeEq(idApInterpreter)))
+
+        "Can interpret an ADT as FreeApplicative operations" {
```

Comment:
Like this but for FreeApplicative instead of Option:
```kotlin
"Cartesian builder should build products over option" {
            Option.map(Option(1), Option("a"), Option(true), { (a, b, c) ->
                "$a $b $c"
            }) shouldBe Option("1 a true")
        }
```
Alternatively we can encode something like that inside the Applicative laws so some basic  builder tests run for all applicatives